proversity_reports_script/report_backend/learning_tracker_report.py

```diff
@@ -0,0 +1,91 @@
+"""
+Report backend for completion report.
+"""
+import csv
+
+import boto3
+
+from proversity_reports_script.report_backend.base import AbstractBaseReportBackend
+
+
+class LearningTrackerReportBackend(AbstractBaseReportBackend):
+    """
+    Backend for Completion report.
+    """
+
+    def __init__(self, *args, **kwargs):
+        extra_data = kwargs.get('extra_data', {})
+        super(LearningTrackerReportBackend, self).__init__(extra_data.get('SPREADSHEET_DATA', {}))
+        amazon_settings = extra_data.get('AWS_DATA', {})
+        self.amazon_bucket = amazon_settings.get('amazon_bucket', '')
+        self.file_prefix = amazon_settings.get('file_prefix', '')
+
+    def generate_report(self, json_report_data):
+        """
+        Main logic to generate the report.
+        """
+        if json_report_data.get('status') != 'SUCCESS':
+            return None
+
+        report_data = json_report_data.get('result', {})
+        report_data.update(self._get_edx_courses_data_from_csv())
+        print(report_data)
+
+    def json_report_to_csv(self, json_report_data):
+        """
+        Process json data to convert into csv format.
+        """
+        raise NotImplementedError()
+
+    def create_csv_file(self, course, body_dict, spreadsheet_id):
+        """
+        Creates the csv file with the passed arguments, and then save it locally.
+        """
+        raise NotImplementedError()
+
+    def upload_file_to_storage(self, course, path_file):
+        """
+        Uploads the csv report, to S3 storage.
+        """
+        raise NotImplementedError()
+
+    def _get_edx_courses_data_from_csv(self):
+        """
+        Gets csv file from S3 and extracts the data.
+        """
+        amazon_resource = boto3.resource('s3')
+        amazon_bucket = amazon_resource.Bucket(self.amazon_bucket)
+        amazon_client = boto3.client('s3')
+        previous = None
+        data = {}
+
+        for amazon_object in amazon_bucket.objects.filter(Prefix=self.file_prefix):
+
+            if not previous or amazon_object.last_modified > previous:
+                key = amazon_object.key
+
+            previous = amazon_object.last_modified
+
+        with open('aws-file.csv', 'wb') as csv_data:
```

Comment:
aws-file.csv should be a constant, since you are using it twice and it's a file name. @diegomillan